src/uu/head/src/head.rs

```diff
@@ -1,240 +1,300 @@
-//  * This file is part of the uutils coreutils package.
-//  *
-//  * (c) Alan Andrade <alan.andradec@gmail.com>
-//  *
-//  * For the full copyright and license information, please view the LICENSE
-//  * file that was distributed with this source code.
-//  *
-//  * Synced with: https://raw.github.com/avsm/src/master/usr.bin/head/head.c
+use std::convert::TryFrom;
+use std::io::{ErrorKind, Read, Seek, SeekFrom, Write};
+use uucore::{crash, executable, show_error};
 
-#[macro_use]
-extern crate uucore;
+mod app;
+mod constants;
+mod parse;
+mod split;
+use app::*;
 
-use std::collections::VecDeque;
-use std::fs::File;
-use std::io::{stdin, BufRead, BufReader, Read};
-use std::path::Path;
-use std::str::from_utf8;
-
-static SYNTAX: &str = "";
-static SUMMARY: &str = "";
-static LONG_HELP: &str = "";
-
-enum FilterMode {
-    Bytes(usize),
-    Lines(usize),
-    NLines(usize),
-}
+fn rbuf_n_bytes(input: &mut impl std::io::BufRead, n: usize) -> std::io::Result<()> {
+    if n == 0 {
+        return Ok(());
+    }
+    let mut readbuf = [0u8; constants::BUF_SIZE];
+    let mut i = 0usize;
 
-struct Settings {
-    mode: FilterMode,
-    verbose: bool,
-    zero_terminated: bool,
-}
+    let stdout = std::io::stdout();
+    let mut stdout = stdout.lock();
 
-impl Default for Settings {
-    fn default() -> Settings {
-        Settings {
-            mode: FilterMode::Lines(10),
-            verbose: false,
-            zero_terminated: false,
+    loop {
+        let read = input.read(&mut readbuf)?;
```

Comment:
This (and basically everywhere else using `read()`) should check for `ErrorKind::Interrupted`.